### DISCUSSION TITLE
BREAKING CHANGE: migration from "trivy" to "trivy image" command

### DISCUSSION THREAD

**Discussion by knqyf263** [aquasecurity/trivy#1515] [Announcements]
# Description
`trivy` command behaves the same as `trivy image` for backward compatibility. But it was deprecated in v0.8.0. It's been more than a year and this option will be removed in v0.23.0 which is supposed to be released at the end of January 2022. Please migrate to `trivy image`.

## 🙅 NG
```
$ trivy [YOUR_IMAGE]
```

## 🙆‍♂️ OK 
```
$ trivy image [YOUR_IMAGE]
```

**Comment by danielhoherd** [comment]
PSA for anybody else who is running into the following errors:

```
FATAL            flag provided but not defined: -cache-dir
Incorrect Usage: flag provided but not defined: -cache-dir
```

`--cache-dir` is a global arg, so you must construct the command as `trivy --cache-dir $CACHE_DIR image ...`

**Comment by rizwan-kh** [comment]
Created a PR #1575 to update the document of integration with GitLab CI

**Comment by caiohasouza** [comment]
Hi,

Perfect, thank you!

Regards

**Comment by PranavBhattarai** [comment]
1 **noob** question though:

I installed trivy rpm package. And run this command: ```trivy -verison``` and I see this line in the end the output:

```2022-04-24T13:58:25.244+0545	FATAL	flag provided but not defined: -verison```

Is this a normal way to say, "hello, welcome to trivy", or what is this? I am guessing this is normal, but the way it looks in the terminal while checking _just version_, makes it feels as if there is a **problem**.

![image](https://user-images.githubusercontent.com/32763112/164967024-8219c7bf-83ce-4688-8950-2c82ae13d8c6.png)

**Reply by knqyf263** [reply]
$ trivy --version
$ trivy -v
$ trivy version

All works other than `trivy -version`

**Reply by PranavBhattarai** [reply]
Ah, such a dumb question. I didn't see that I made a mistake in typing "version". 
Whenever the user gives wrong input, I expected 1 liner warning/error message. 
In this case, the very **first line** and **last line** was telling me _where_ I made mistake. The contents of ```trivy --help``` in between those two warnings, confused me.
